### PR TITLE
Do not return if data is unchanged

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-698

### CODE REVIEW

File: manager/dedup.go [hunk 1]
```diff
@@ -342,12 +353,23 @@ START:
...
+	}
+
+	if meta.LastIndex < lastIndex {
+		log.Printf("[TRACE] (dedup) %s had a lower index, resetting", path)
+		lastIndex = 0
+		goto START
+	}
+	lastIndex = meta.LastIndex
+
+	if reflect.DeepEqual(pair, lastPair) {
```
Comment (slackpad) [inline manager/dedup.go]: Did you find that you needed this? The index check should be sufficient and much cheaper.

Comment (sethvargo) [inline manager/dedup.go]: I needed that when I wrote the poller on the original Consul Template. I found the index could change if someone updated the key to the same value.

Comment (sethvargo) [inline manager/dedup.go]: Is that still the case?

Comment (slackpad) [inline manager/dedup.go]: That's still true, though in that case the `DeepEqual` will see the index changes - you might want to just compare the values of the keys.

**Pull request merged**